### PR TITLE
Initial 1.4.0 release notes

### PR DESCRIPTION
**Pull request by howardjohn**
There will definitely be more release notes to add in the future, but
for now I think we can focus on getting a few in and iterate from there
to try to avoid the 300 comment PR madness.

### CODE REVIEW

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 1]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
 
-TBD
+## Traffic management
+
+- **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).
+- **Improved** the Envoy sidecar to exit upon crashing, giving better visibility of health.
+- **Improved** Pilot to skip sending redundant configuration to Envoy when no changes are required.
+
+## Security
+
```
Comment (yangminzhu) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]: Can we add:
```
- **Added** v1beta1 authorization policy for access control which deprecates the v1alpha1 RBAC policy.
```

Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: @yangminzhu definitely needs to be added at some point, I was waiting for https://github.com/istio/istio.io/issues/5289 before adding this. Unlike previous release notes PRs I am trying to split this up and just get the notes that are ready in asap and we can continue to add others as they are ready

**Comment by howardjohn** [comment]
@istio/wg-docs-maintainers-english can you please take a look?

File: content/en/news/2019/announcing-1.4/change-notes/index.md (hunk 1 above)
Comment (yangminzhu) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: Sounds good, we can update this later when other authz-v2 docs are merged.

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 2]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Added** OpenAPI v3 schema validation to CRDs. Please see the [Upgrade Notes](/news/2019/announcing-1.4/upgrade-notes) for details.
+
+## Installation
+
+- **Remove** the `proxy_init` docker image. Instead, the `istio-init` container will reuse the `proxyv2` image
+- **Update** the base image to `ubunutu:bionic`.
+
+## `istioctl`
+
+- **Added** [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to wait until an Istio configuration has been pushed to all Envoy sidecars.
```
Comment (rcaballeromx) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Added** the experimental sub-command [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to have Istio wait until it has pushed a configuration to all Envoy sidecars.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 3]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Added** [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) to help manage Istio across multiple clusters.
```
Comment (rcaballeromx) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Added** the experimental [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) sub-command to help manage Istio across multiple clusters.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 4]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Added** [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) to manage webhook configuration.
```
Comment (rcaballeromx) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Added** the experimental [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) sub-command to manage webhook configuration.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 5]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Improved** [`istioctl version`](/docs/reference/commands/istioctl/#istioctl-version) to show Envoy proxy versions.
```
Comment (rcaballeromx) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Improved** the [`istioctl version`](/docs/reference/commands/istioctl/#istioctl-version) sub-command. It now shows the Envoy proxy versions.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 6]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
 weight: 10
 ---
 
-TBD
+## Traffic management
+
+- **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).
+- **Improved** the Envoy sidecar to exit upon crashing, giving better visibility of health.
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: Not really.. previously it was completely obscured, now its just standard kubernetes. if it gets OOM killed kubernetes will mark it as OOMKilled. If it is crashing it will be marked as CrashLoopBackoff. Its both self explanatory and part of basic kubernetes docs, so I don't think we need to specifically document it on our tasks

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 7]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
 weight: 10
 ---
 
-TBD
+## Traffic management
+
+- **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: I am not quite sure what you mean. https://preliminary.istio.io/docs/concepts/traffic-management/ does not have anything about mirroring. We already have a doc that explains this, and I linked to it. Its not really a "different type" of mirroring, its just sending only x% of traffic instead of 100%. Would it make sense to just rephrase this to "Added support for mirroring a percentage of traffic"?

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 8]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+## Telemetry
+
+- **Added** support for configuring [stat patterns](/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig) for Envoy stats.
+- **Added** prefix to Envoy HTTP stats to specify traffic direction (`inbound` or `outbound`).
+
+## Configuration management
+
+- **Added** many additional validation checks to [`istioctl analyze`](/docs/ops/diagnostic-tools/istioctl-analyze/).
+- **Added** experimental support for validation message in Istio [resource statuses](/docs/ops/diagnostic-tools/istioctl-analyze/#enabling-validation-messages-for-resource-status).
+- **Added** OpenAPI v3 schema validation to CRDs. Please see the [Upgrade Notes](/news/2019/announcing-1.4/upgrade-notes) for details.
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: https://github.com/istio/istio.io/pull/5174 (already merged)

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 9]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+
+## Installation
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: Yes, I added only the portion of release notes that were ready. I thought it would be easier to split this rather than have one mega-pr

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 10]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+
+- **Remove** the `proxy_init` docker image. Instead, the `istio-init` container will reuse the `proxyv2` image
+- **Update** the base image to `ubunutu:bionic`.
+
+## `istioctl`
+
+- **Added** [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to wait until an Istio configuration has been pushed to all Envoy sidecars.
+- **Added** [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) to help manage Istio across multiple clusters.
+- **Added** [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) to manage webhook configuration.
```
Comment (ostromart) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]: - **Added** [`istioctl experimental upgrade`](/docs/reference/commands/istioctl/#istioctl-experimental-upgrade) to upgrade Istio installed with `istioctl manifest` commands.

File: content/en/news/2019/announcing-1.4/change-notes/index.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ostromart) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: - **Added** [operator controller](/docs/ops/setup/standalone-operator/) for dynamic updates to Istio installation based on a user CustomResourceDefinition.

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 11]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Added** the experimental option to enable validation messages for Istio [resource statuses](/docs/ops/diagnostic-tools/istioctl-analyze/#enabling-validation-messages-for-resource-status).
+- **Added** OpenAPI v3 schema validation of Custom Resource Definitions (CRDs). Please see the [Upgrade Notes](/news/2019/announcing-1.4/upgrade-notes) for details.
+
+## Installation
+
+- **Removed** the `proxy_init` Docker image. Instead, the `istio-init` container reuses the `proxyv2` image.
+- **Updated** the base image to `ubunutu:bionic`.
+
+## `istioctl`
+
```
Comment (elfinhe) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]: Please also add one line for upgrade command:
```
- **Added** the experimental sub-command [`istioctl experimental upgrade`](docs/setup/upgrade/istioctl-upgrade/) to performs an in-place upgrade of the Istio control plane with eligibility checks.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (rcaballeromx) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: Yes that would be enough clarity.
The correct link for the mirroring task is this one: `https://istio.io/docs/tasks/traffic-management/mirroring/`

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 12]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
-TBD
+## Traffic management
+
+- **Added** support for fractional [request mirroring](/docs/tasks/traffic-management/mirroring/).
+- **Improved** the Envoy sidecar. The Envoy sidecar now exits when it crashes. This change makes it easier to see whether or not the Envoy sidecar is healthy.
+- **Improved** Pilot to skip sending redundant configuration to Envoy when no changes are required.
+
+## Security
+
+- **Improved**Citadel to periodically check and rotate the expired root certificate when running in self-sign CA mode.
```
Comment (rcaballeromx) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Improved** Citadel to periodically check and rotate the expired root certificate when running in self-sign CA mode.
```

**Comment by howardjohn** [comment]
Will add the extra notes in a follow up PR

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 13]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+## `istioctl`
+
+- **Added** the experimental sub-command [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to have Istio wait until it has pushed a configuration to all Envoy sidecars.
+- **Added** the experimental [`istioctl experimental mulitcluster`](/docs/reference/commands/istioctl/#istioctl-experimental-multicluster) sub-command to help manage Istio across multiple clusters.
+- **Added** the experimental [`istioctl experimental post-install webhook`](/docs/reference/commands/istioctl/#istioctl-experimental-post-install-webhook) sub-command to manage webhook configuration.
+- **Improved** the [`istioctl version`](/docs/reference/commands/istioctl/#istioctl-version) sub-command. It now shows the Envoy proxy versions.
+
+## Miscellaneous
+
+- **Replaced** daily builds of Istio with builds for [each commit](https://github.com/istio/istio/wiki/Dev%20Builds).
```
Comment (geeknoid) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]: This doesn't belong in customer-facing release notes.

File: content/en/news/2019/announcing-1.4/change-notes/index.md [hunk 14]
```diff
@@ -4,4 +4,39 @@ description: Istio 1.4 release notes.
...
+- **Added** OpenAPI v3 schema validation of Custom Resource Definitions (CRDs). Please see the [Upgrade Notes](/news/2019/announcing-1.4/upgrade-notes) for details.
+
+## Installation
+
+- **Removed** the `proxy_init` Docker image. Instead, the `istio-init` container reuses the `proxyv2` image.
+- **Updated** the base image to `ubunutu:bionic`.
+
+## `istioctl`
+
+- **Added** the experimental sub-command [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) to have Istio wait until it has pushed a configuration to all Envoy sidecars.
```
Comment (geeknoid) [inline content/en/news/2019/announcing-1.4/change-notes/index.md]:
```suggestion
- **Added** the experimental [`istioctl experimental wait`](/docs/reference/commands/istioctl/#istioctl-experimental-wait) sub-command to have Istio wait until it has pushed a configuration to all Envoy sidecars.
```

File: content/en/news/2019/announcing-1.4/change-notes/index.md (hunk 13 above)
Comment (howardjohn) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: I am not sure I agree with this. I have had a number of users ask where they can find these builds. It's at the bottom of the notes, is it harmful to have this here?

Comment (geeknoid) [reply content/en/news/2019/announcing-1.4/change-notes/index.md]: It's not harmful, but it goes against the general direction we're trying to follow of being customer-focused rather than developer/hacker-focused.

**Comment by howardjohn** [comment]
Ok, I will remove it. I didn't realize we were moving in that direction,
thanks!

**Comment by howardjohn** [comment]
/retest

seems like a flake
